File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/Models/Document.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankingPaymentsApp_API.Models
{
    public class Document
    {
        [Key]
        public int DocumentId {  get; set; }
        [Required(ErrorMessage ="DocumentURL is Required!")]
        public string DocumentURL {  get; set; }
        [Required(ErrorMessage = "Document Name is Required!")]
        public string DocumentName { get; set; }
        [Required(ErrorMessage = "Document Type is Required!")]
        [ForeignKey("ProofType")]
        public int ProofTypeId { get; set; }
        public virtual ProofType? ProofType { get; set; }

        public string PublicId { get; set; }

        [ForeignKey("Account")]
        public int ClientId { get; set; }
        public virtual ClientUser? Account { get; set; }
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/Models/Document.ts
import { ClientUser } from "./ClientUser";
import { ProofType } from "./ProofType";

export interface Document {
  documentId?: number;
  documentURL?: string;
  documentName?: string;
  proofTypeId?: number;
  publicId?: string;
  clientId?: number;
  client?: ClientUser;
  proofType?: ProofType;
}